README.md

```diff
@@ -1 +1,268 @@
-# react-file-upload
\ No newline at end of file
+# File Management with Preview - Fully Customized
+
+A versatile and user-friendly file management system built with React, Next.js and TypeScript that allows for single and multiple file uploading with a preview feature. It allows you to select files and preview them, returning an array of selected files. It allows customizing design by overriding the style classes.
+
+<img src="./gifs/full.gif"/>
+
+Checkout the live demo on, codesandbox <br />
+[![codesandbox.io](https://codesandbox.io/favicon.ico)](https://codesandbox.io/p/sandbox/cranky-breeze-r4hht7?file=%2Fsrc%2Fmain.js)
+
+---
+
+## Table of Contents
+
+- [Features](#features)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+- [Examples](#examples)
+  - [Canvas View](#canvas-view)
+  - [Square View](#square-view)
+  - [Horizontal Long Square View](#horizontal-long-square-view)
+  - [Circular View](#circular-view)
+  - [Over-ride CSS](#over-ride-css)
+- [Properties and Events](#properties-and-events)
+- [Usage](#usage)
+- [Contributing](#contributing)
+- [License](#license)
+- [Contact Information](#contact-information)
+
+---
+
+## Features
+
+- **Single File Upload:** Users can upload or change a single file.
+- **Multiple File Management:** Easily manage multiple files.
+- **File Preview:** Provides a preview of uploaded files (e.g., images, videos, gifs).
+- **Responsive Design:** Ensures a seamless experience on various devices.
+- **Fully customized:** Customize file upload UI as per your requirements
+
+---
+
+## Getting Started
+
+Follow below instructions to configure this package into your project.
+
+### Prerequisites
+
+Before you begin, make sure you have the following software installed:
+
+- [Node.js](https://nodejs.org/) - v20.x
+- React with Next.js
+- sass
+
+### Installation
+
+Using npm:
+
+```
+npm install @canopassoftware/vue-file-upload
+```
+
+Using yarn:
+
+```
+yarn add @canopassoftware/vue-file-upload
+```
+
+---
+
+## Examples
+
+We are providing some examples with design. so, you can easily take it and use into your project.
```

Comment:
We are providing some examples with design. so, you can easily use it in your project.